packages/dds/merge-tree/src/test/obliterate.deltaCallback.spec.ts

```diff
@@ -0,0 +1,165 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { MergeTreeDeltaType } from "../ops.js";
+import { MergeTreeDeltaCallback } from "../mergeTreeDeltaCallback.js";
+import { useStrictPartialLengthChecks } from "./testUtils.js";
+import { ReconnectTestHelper } from "./reconnectHelper.js";
+
+describe("obliterate delta callback", () => {
+	useStrictPartialLengthChecks();
+
+	let length: number;
+	let cb: MergeTreeDeltaCallback;
+
+	beforeEach(() => {
+		length = 0;
+		cb = (opArgs, deltaArgs) => {
+			switch (opArgs.op.type) {
+				case MergeTreeDeltaType.INSERT: {
+					for (const { segment } of deltaArgs.deltaSegments) {
+						length += segment.cachedLength;
+					}
+					break;
+				}
+				case MergeTreeDeltaType.REMOVE:
+				case MergeTreeDeltaType.OBLITERATE: {
+					for (const { segment } of deltaArgs.deltaSegments) {
+						length -= segment.cachedLength;
+					}
+					break;
+				}
+				default: {
+				}
+			}
+		};
+	});
+
+	it("works", () => {
```

Comment:
small nit: I probably would have tried to write this test so that the comments you have on `local`/`remote` are in the test case name, so something like

```
'obliterate delta callback'
	'is invoked'
		'on local obliterate'
		'on remote obliterate'
	'is mutually exclusive with overlapping remove'
		'when the obliterate is first'
		'when the remove is first'
```
for test suite / case structure. Or something like that. Under this structure the insert case would more naturally go under the insert callback suite, but putting it here in a different top-level suite seems fine too.

We're not super consistent about this sort of organization so I'm not super picky when reviewing, but I find writing test cases like this tends to help preserve the intent of the test case and also make refactors easier for people touching the test case later. Something to think about :)

Reply:
another advantage I've found for naming test cases like this is it lends itself well to LLMs--copilot is quite good at writing the correct code in many cases when given somewhat explicit directions like this.